### ISSUE TITLE
Vendor earnings remain unchanged when I, as an admin, have processed a refund on an order.

### ISSUE THREAD

**Issue by bkluiter** [getdokan/dokan#2091]
### Bug Description

Vendor earnings remain unchanged when I, as an admin, have processed a refund on an order.

I am using Dokan 3.9.2. (lite) WordPress and WooCommerce are all up to date. I have turned off all other plugins and even all themes. The problem persists. In the Vendor's dashboard, it is clearly visible that the money has been refunded and the order is marked as refunded. However, the earnings for the vendor remain the same. It is also possible for the vendor to request a payout of this value. Something is going wrong, but I can't figure out where.

Who can help me?

### Step To Reproduce

In the admin section, process a refund on an order so that the customer gets their money back and the status of the order is changed to refunded.

### Expected Behaviour

The earnings of a vendor should be reduced by the value of the refunded amount to the customer after an order is refunded.

### Actual Behaviour

The earnings for the vendor stays the same.

### Additional Information

No

### Environment (please complete the following information)

```shell
1. Dokan Version: 3.9.2
2. Dokan Pro Version: -
3. WooCommerce Version: 8.3.1
4. Browser: Chrome
5. Current Theme: Blocksy
```

**Comment by bkluiter** [comment]
Am I the only one experiencing this issue? The problem also arises when you do a partial refund. The solution provided by @alamgircsebd does not work in this case either.

**Comment by nurul-umbhiya** [comment]
@bkluiter Refunds are a feature of Dokan Pro. You need to install and activate Dokan Pro to enable the refund functionality.
Also, if you are using Dokan Pro, kindly try to refund from the line items, not from the `Refund Amount` field directly to initiate a refund.

**Comment by bkluiter** [comment]
@nurul-umbhiya I have also tested it with our Pro license, and the same problem exists there. In the case of a partial refund of an order (for example, if a customer returns 1 out of 2 ordered products), after processing the return, you indicate that the product needs to be refunded. The revenue, and therefore the payable amount, for the vendor as well as the admin fee should then be reduced. However, this is not happening currently. I believe Dokan calculates this only based on a fully refunded order, thereby also adjusting the order status. If a partial refund has occurred, the order status remains marked as completed. For the refunded product, WooCommerce generates a new refund ID. Could it be that Dokan does not take this into account?

**Comment by nurul-umbhiya** [comment]
@bkluiter I would suggest you create a support ticket for this. Kindly add a reference to this GitHub Issue.

**Comment by devAsadNur** [comment]
@bkluiter Refunds should be made from the line items, not from the `Refund Amount` field directly. If we make refund from line items, the vendor earning calculation and all other calculations work fine. I've just checked out with the latest `Dokan Lite`, `Dokan Pro`, and `WooCommerce`.

I'm closing this issue right now. If you still find an issue with this, feel free to create a support ticket. Thanks.

Tested Environment:
--------------------
Dokan Lite: `3.10.1`
Dokan Pro: `3.10.1`
WooCommerce: `8.6.1`
